core/server/common/src/main/java/alluxio/master/journal/DefaultJournalMaster.java

```diff
@@ -75,4 +86,12 @@ public GetTransferLeaderMessagePResponse getTransferLeaderMessage(String transfe
            .setTransMsg(((RaftJournalSystem) mJournalSystem).getTransferLeaderMessage(transferId))
            .build();
   }
+
+  @Override
+  public GetNodeStatePResponse getNodeState() {
+    return GetNodeStatePResponse.newBuilder()
+        .setNodeState(mPrimarySelector.getState() == PrimarySelector.State.PRIMARY
```

Comment:
are you feel good with the approach of getting the node state from the PrimarySelector? I feel this is the best way because it works for both UFS and embedded journals regardless if HA mode is enabled or not.